src/Api/User.hs

```diff
@@ -34,7 +35,7 @@ allUsers =
 -- | Returns a user by name or throws a 404 error.
 singleUser :: String -> App (Entity User)
 singleUser str = do
-    maybeUser <- runDb (selectFirst [UserName ==. str] [])
+    maybeUser <- runDb (selectFirst [UserName ==. (pack str)] [])
```

Comment:
Yeah, I would expect the API definition switching to `Text` would be fine. Can you post the errors you receive?

Reply:
Hi @parsonsmatt. So I undid `(pack str)` to simply `str` (as it was before), and I changed `Capture "name" String` to `Capture "name" Text`. Doing so gave me the [following errors](https://gist.github.com/StevenXL/1f648c0745b3ea59e51b85d7219316dc). 

My understanding of the second error (L38) is that the `str` value is of type `String`, which is surprising since I thought that `Capture "name" Text` would ensure `str` is of type `Text`, given the change to the API. 

The second error I, frankly, couldn't decipher. 

The `User.hs` file I was using is [here])(https://gist.github.com/StevenXL/fb10f26779704eed5e95980b6692b702) for your reference.

Reply:
Ah, right!

So, with an API type like:

```haskell
...
  :<|> "users" :> Capture "name" String :> Get '[JSON] (Entity User)
```

then the function that gets matched to it needs to have the same type. So we can change that `String` to `Text`. Then, when we're putting the API function together, we have to *also* change the type of the function that handles the `singleUser` to use `Text` instead of `String`.

Reply:
Ah - I see. `singleUser` the handler didn't have the signature that we said it was going to have when we declared the `UserAPI` type alias. Thank you!